### PR TITLE
refactor: replace bitwise calculation with calculateBits()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**
Since (bits, perm) => bits... is used very often and there is a function that handles that I have replaced them with the function.

Changes have not been tested right now. Still in progress.

**Status**

- [ ] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by itohatweb** [comment]
Or just close #242 and continue here

File: src/handlers/channel.ts [hunk 1]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
...
+) {
+  const hasManageChannelPerm = await botHasChannelPermissions(
+    channelID,
+    ["MANAGE_CHANNELS"],
+  );
+  if (!hasManageChannelPerm) {
+    throw new Error(Errors.MISSING_MANAGE_CHANNELS);
+  }
+
+  const { allow, deny, ...info } = overwrite;
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:437]: I'm not really sure how necessary this is. But whenever you can avoid making variables its a really good thing. Variables should only be done when you have to reuse them multiple times imo.  I only see these being used once.

In JS, any variable like this takes up space and until Garbage Collector runs this is wasted space. It's also quite performance heavy to create when we could just do `overwrite.allow` below.

File: src/handlers/channel.ts [hunk 2]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
...
+  let channel = cache.channels.get(channelID);
+
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:439]: empty line?

File: src/handlers/channel.ts [hunk 3]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
...
+  const payload = {
+    permission_overwrites: [
+      ...(channel?.permissionOverwrites || []).map((rawOverwrite) => ({
+        id: rawOverwrite.id,
+        type: rawOverwrite.type,
+        allow: rawOverwrite.allow,
+        deny: rawOverwrite.deny,
+      })),
+      {
+        ...info,
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:451]:
```suggestion
        ...overwrite,
```

File: src/handlers/channel.ts [hunk 4]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
...
+        allow: calculateBits(allow),
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:452]:
```suggestion
        allow: calculateBits(overwrite.allow),
```

File: src/handlers/channel.ts [hunk 5]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
...
+        deny: calculateBits(deny),
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:453]:
```suggestion
        deny: calculateBits(overwrite.deny),
```

File: src/handlers/channel.ts [hunk 6]
```diff
@@ -422,6 +421,49 @@ export async function editChannel(
   );
 }
 
+export async function editChannelOverwrite(
+  channelID: string,
+  overwrite: Overwrite,
+  reason?: string,
+) {
+  const hasManageChannelPerm = await botHasChannelPermissions(
```
Comment (Skillz4Killz) [inline src/handlers/channel.ts:429]: This needs a little bit of research as im not 100% certain but if i had to guess, im sure Discord would not let a bot give/remove permission from another role/member if itself did not have the permissions being edited.

Also can you double check MANAGE_ROLES is not required? I think thats the MANAGE Permissions one and im almost positive that might be required

**Comment by Skillz4Killz** [comment]
LOVE THE BITWISE REFACTORING!

File: src/handlers/channel.ts (hunk 6 above)
Comment (itohatweb) [reply src/handlers/channel.ts:429]: Found out that permissions are little more complex. Other functions are facing the same problem too.

File: src/handlers/channel.ts (hunk 1 above)
Comment (itohatweb) [reply src/handlers/channel.ts]: Removed that

**Comment by ayntee** [comment]
Can you remove irrelevant changes from the PR? (editChannelOverwrite())

**Review by ayntee — APPROVED** [review]
LGTM! Can we undraft this PR?

**Comment by itohatweb** [comment]
Yes, editChannelOverwrite() should be in #242

**Pull request merged**